From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: vanhofen <vanhofen@gmx.de>
Date: Wed, 28 Oct 2026 22:45:00 +0100
Subject: [PATCH] configure.ac: allow autoconf 2.69

Python 3.13+ requires autoconf 2.72 to regenerate configure. The
buildsystem regenerates configure with the build host's autoconf,
and common build hosts still ship older versions, e.g. autoconf
2.71 (Debian 12, Ubuntu 22.04/24.04) or 2.69 (Debian 11,
Ubuntu 20.04). Both handle configure.ac just fine.

Differences of a configure generated by autoconf 2.69:
- AC_USE_SYSTEM_EXTENSIONS doesn't define the newer
  __STDC_WANT_*__ macros. On Linux _GNU_SOURCE is still defined,
  which exposes the same declarations.
- The -Wstrict-prototypes check fails because autoconf 2.69 emits
  "int main ()", so this warning flag isn't added to CFLAGS_NODIST.

Buildroot does not need this, as it uses its own host-autoconf.
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index f51935d..d5fb086 100644
--- a/configure.ac
+++ b/configure.ac
@@ -12,7 +12,7 @@ dnl
 # Set VERSION so we only need to edit in one place (i.e., here)
 m4_define([PYTHON_VERSION], [3.14])
 
-AC_PREREQ([2.72])
+AC_PREREQ([2.69])
 
 AC_INIT([python],[PYTHON_VERSION],[https://github.com/python/cpython/issues/])
 
